include/zephyr/pm/state.h

```diff
@@ -339,6 +347,18 @@ struct pm_state_info {
  */
 uint8_t pm_state_cpu_get_all(uint8_t cpu, const struct pm_state_info **states);
 
+/**
+ * Retrieve the element index of a CPU's power state.
+ *
+ * @param cpu CPU index.
+ * @param state	Power state.
+ * @param substate_id Power state substate id.
+ *
+ * @return Index of the pm_state_info element for a CPU's (sub-)state.
+ */
+uint8_t pm_state_cpu_get_index(uint8_t cpu, enum pm_state state,
+			       uint8_t substate_id);
+
```

Comment:
I don't think this needs to be part of the API tbh, if really needed this could probably be internal to where it is needed.

Reply:
I thought it might be interesting to have. The longer I looked at usage of `z_cpus_pm_state[id].state`, the more I thought it might be a better solution to have a pointer to a power state array member inside it instead of just an unlinked `pm_state_info` instance.

Reply:
Removed, as no longer applicable.

Reply:
This function should return a signed int since it returns a negative value in case of error, otherwise we will have -1 wrongly promoted to a positive value.

Also, my point is, that is a very very specific function, lets not include it in a public header, just remove this declaration and declare it in  `pm.c`

``extern int8_t pm_state_get_index(enum pm_state state, uint8_t substate_id);``